### PR TITLE
[incubator-kie-drools#5728] Update PR template

### PR DESCRIPTION
**Pull request by tkobayas**
**Issue**:

* https://github.com/apache/incubator-kie-drools/issues/5728

1. Remove old kiegroup specific guidance
  -> Removed Jenkins command comment which is no longer working.

2. Suggest to use github issue rather than JIRA

### CODE REVIEW

**Comment by tkobayas** [comment]
@mariofusco @rgdoliveira @baldimir Please review, thanks!

**Review by mariofusco — APPROVED** [review]

**Review by rgdoliveira — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
 **Thank you for submitting this pull request**
 
-**NOTE!:** Double check the target branch for this PR.
+**NOTE!:** Double-check the target branch for this PR.
 The default is `main` so it will target Drools 8 / Kogito.
-If this PR is not strictly related to drools and kogito project in `drools.git`, it should probably target `7.x`as a branch
+If this PR is targeted to `7.x`, this should be sent to `kiegroup/drools` repository, instead of `apache/incubator-kie-drools` repository.
```
Comment (baldimir) [inline .github/pull_request_template.md]: I think we could skip/delete this, as I am not sure if it is ok to have these kind of recommendations to forks, in an Apache project.

**Review by baldimir — COMMENTED** [review]
Hi @tkobayas, thank you for this. I added one very minor comment. What do you think about it please?

File: .github/pull_request_template.md (hunk 1 above)
Comment (tkobayas) [reply .github/pull_request_template.md]: @baldimir Thanks, agreed. I removed the mention to kiegroup repo.

**Review by baldimir — APPROVED** [review]

**Pull request merged**